src/main/java/wooteco/subway/dto/PathResponse.java

```diff
@@ -0,0 +1,41 @@
+package wooteco.subway.dto;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import wooteco.subway.domain.ShortestPath;
+
+public class PathResponse {
+
+    private final List<StationResponse> stations;
+    private final int distance;
+    private final int fare;
+
+    private PathResponse() {
+        this(null, 0, 0);
+    }
```

Comment:
이 생성자는 필요한가요?

Reply:
jackson에서 DTO를 객체로 매핑할 때 사용하는 기본생성자를 이용하는 것을 지난 미션에서 오류를 통해 발견했습니다.
Response일 때는 상관이 없을 수 있지만 DTO에는 무조건 빈 생성자를 만들어주어야겠다는 생각으로 구현하고 있습니다.
생각해보니 필드에 final 키워드를 제외하고 빈 생성자로 두는 것이 더 괜찮아보여 수정하였습니다!
```suggestion

    private PathResponse() {
    }
```

Reply:
네 혹시 DTO에 언제 기본생성자가 필요한지도 공부해보셨나요?
Response DTO에는 기본생성자가 왜 없어도 동작할까요?
기본생성자가 없어도 문제 없다면 삭제해주는게 좋을 것 같은데, `DTO에는 무조건 빈 생성자를 만들어주어야겠다`라고 생각하신 이유가 있을까요? 언제 기본생성자가 필요한지를 안다면 필요할때만 쓸 수 있을 것 같네요 :)

Reply:
소니의 말을 듣고 학습해보았어요! 😁

보통 기본 생성자가 필요한 경우는 라이브러리 내부에서 리플렉션을 이용할 때입니다.
리플렉션은 구체적인 클래스 타입을 몰라도 런타임에 클래스에 대한 정보를 알아낼 수 있는 기능을 제공하는 API인데 여기서 함정은 생성자의 인자를 알아낼 수 없습니다. 프라이빗 메서드를 퍼블릭으로 바꾸는 등 막강한 기능을 가짐에도 이런 허점이 있죠. 그렇기 때문에 기본 생성자를 필요로 합니다.

Jackson 에서는 mapper 바인딩을 할 때 내부에서 리플렉션을 이용합니다. 그렇기 때문에 기본 생성자를 필요로 합니다.

제가 `DTO에 무조건 빈 생성자를 만들어주어야겠다` 라고 생각한 이유는 빈 생성자를 private으로 생성한다면 프로덕션 코드에 그렇게 큰 영향을 받지 않는다고 생각하고 있고 
 비용도 그렇게 크지 않다고 생각합니다. 또한 Response 객체라도 ObjectMapper를 이용할 수 있다고 봅니다.

하지만 선택적으로 빈 생성자를 넣어주다 보면 이후 실수로 DTO에 빈 생성자를 빠뜨릴 수 있다봅니다. 그러면 Jackson과 같은 라이브러리에서 해당 생성자의 인자를 알지 못해 생성하지 못하는 오류가 뜰 가능성이 생깁니다.

그래서 저는 `DTO에는 무조건 빈 생성자를 만들어준다` 라는 생각을 따르고 있습니다.